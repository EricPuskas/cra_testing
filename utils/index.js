import checkPropTypes from "check-prop-types";
import { applyMiddleware, createStore } from "redux";
import rootReducer from "../src/reducers";
import { middleware } from "../src/store";

export const findByAttribute = (component, attr) =>
  component.find(`[data-test="${attr}"]`);

export const checkProps = (component, expectedProps) =>
  checkPropTypes(component.propTypes, expectedProps, "props", component.name);

export const testStore = initState => {
  const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
  return createStoreWithMiddleware(rootReducer, initState);
};
