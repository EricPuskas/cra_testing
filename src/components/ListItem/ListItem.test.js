import React from "react";
import { shallow } from "enzyme";
import { findByAttribute, checkProps } from "../../../utils";
import ListItem from "./ListItem";

describe("ListItem Component", () => {
  describe("Checking PropTypes", () => {
    it("Should not throw a warning", () => {
      const expectedProps = {
        title: "Example Title",
        desc: "Example Description"
      };
      const propsError = checkProps(ListItem, expectedProps);
      expect(propsError).toBeUndefined();
    });
  });
});

describe("Renders without crashing", () => {
  let component;
  beforeEach(() => {
    const props = {
      title: "Example Title",
      desc: "Example Description"
    };
    component = shallow(<ListItem {...props} />);
  });
  it("Should pass snapshot test", () => {
    expect(component.debug()).toMatchSnapshot();
  });

  it("Should render without error", () => {
    let elem = findByAttribute(component, "listItem");
    expect(elem.length).toBe(1);
  });

  it("Should render a title", () => {
    const title = findByAttribute(component, "title");
    expect(title.length).toBe(1);
  });

  it("Should render a description", () => {
    const desc = findByAttribute(component, "desc");
    expect(desc.length).toBe(1);
  });
});

describe("Should not render", () => {
  let component;
  beforeEach(() => {
    const props = {
      desc: "Example Description"
    };
    component = shallow(<ListItem {...props} />);
  });
  it("Should not render h2 when no title prop", () => {
    const elem = findByAttribute(component, "listItem");
    expect(elem.length).toBe(0);
  });
});
