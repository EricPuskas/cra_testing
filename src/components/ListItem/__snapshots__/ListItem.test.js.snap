// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing Should pass snapshot test 1`] = `
"<div data-test=\\"listItem\\">
  <h2 data-test=\\"title\\">
    Example Title
  </h2>
  <p data-test=\\"desc\\">
     
    Example Description
     
  </p>
</div>"
`;
